utils/aplay/aplay.c

```diff
@@ -1139,6 +1174,21 @@ int main(int argc, char *argv[]) {
 		return EXIT_FAILURE;
 	}
 
+	if (strcmp(volume_type_str, "mixer") == 0)
+		volume_type = VOL_TYPE_MIXER;
+	else if (strcmp(volume_type_str, "software") == 0)
+		volume_type = VOL_TYPE_SOFTWARE;
+	else if (strcmp(volume_type_str, "none") == 0)
+		volume_type = VOL_TYPE_NONE;
+	else if (strcmp(volume_type_str, "auto") == 0)
+		volume_type = VOL_TYPE_AUTO;
+	else {
+		fprintf(stderr, "Invalid volume control type '%s'. Try '%s --help' for more information.\n", volume_type_str, argv[0]);
```

Comment:
How about printing the error message in a similar way `bluealsa` does it? The user will get the info about correct values right away. Also (IMHO) it will simplify autocompletion for this option.